### DISCUSSION TITLE
Will v1.0.0 be a stable release of Nushell?

### DISCUSSION THREAD

**Discussion by trymeouteh** [nushell/nushell#14102] [Q&A]
To my understanding Nushell is in the alpha stage and just relesed v0.99.0. Once v1.0.0 is released, will this be the first stable release of Nushell?

**Comment by fdncred** [comment]
When 1.0 is released, it should be the first stable version of nushell. However, I believe we plan to go to 0.100.0 after 0.99.0.

**Reply by andriykrefer** [reply]
Any expected date to v1?

**Reply by fdncred** [reply]
No, because it's dependent on when some pieces come together that is dependent on the amount of available time that contributors have.

**Reply by NotTheDr01ds** [reply]
Keep in mind that even pre-1.0, Nushell is *fairly* stable, and many folks use it as their default shell already.  Updates are every 4 weeks, and breaking changes are usually well-documented).  Breaking changes are a "good new; bad news" thing, since they typically *improve* things ;-).

Side-note: Given the timing of your question, I'm guessing there's a reasonable chance you might have come here after watching [this popular Nushell video](https://youtu.be/uJsZATwQ3R8?si=l5nxy9OUdmhmKPKZ)?

**Reply by andriykrefer** [reply]
I am using nushell for a while, with some build scripts, and I "locked" the version I used in the scripts. Every time I update the version, something breaks. And that is not a surprise, since I am aware it is not supposed to be stable yet.
For shell usage, these little changes may not be a problem, but for scripts, being _actually_ stable is very desirable.

About the video: I did not watch, I just searched for "stable release" in the discussion and picked the most recent related thread.

**Reply by ghost** [reply]
> Every time I update the version, something breaks.

you can reduce these issues drastically by beeing really explicit. for example:
* use brackets everywhere (`let foo = ((true) and ($bar == "xyz"))`)
* escape ALL external commands (`^git`)
* use quotes, etc wherever allowed (`print "hi"`)
* use long forms of parameters for better error messages (`str length --grapheme-clusters`) and specify the "default" option even if not needed
* etc

while it has been really bad during the 0.80s its now gotten pretty rare for my scripts to break.

i also started testing around with a "dependency" you can just import at the top of your script to make it forward-compatible: <https://github.com/Jan9103/nine> (but for now its more of a test)

**Reply by andriykrefer** [reply]
Thanks for the tips @Jan9103!
